.eslintrc.json

```diff
@@ -5,15 +5,17 @@
         "node": true
     },
     "parserOptions": {
-        "ecmaVersion": 6
+        "ecmaVersion": 2018
```

Comment:
Needed for Unicode property escapes in regexps: `/\p{L}/u`.

This requires Node.js 10 or later – which is why I’ve removed Node.js 8 from supported versions in other files. Node.js 8 is EOL anyway so I think it should be fine.

Reply:
My policy (about which I could be persuaded) is that dropping node 8 needs to wait till elm 0.19.2 when it comes. Out of interest: why do we need to regex for unicode?

Reply:
To be able to match variable names. I’ll show where in the code this is.

Reply:
Either way, if supporting Node.js 8 is desired we could use a worse regex on Node.js 8, which means Node.js 8 at least gets partial support.

Reply:
Linking to https://github.com/rtfeldman/node-test-runner/issues/416 for reference.

Note that Node.js 8 became EOL 10 months ago – 2019-12-31.

Supporting EOL Node.js versions sounds like a bad idea to me. Elm 0.19.2 might be years away. By then, at least Node.js 10 will be EOL as well (2021-04-30). Supported Elm version and supported Node.js version has no connection in my mind.

- Upgrading Node.js is super easy these days anyway, IMO.
- And if someone doesn’t want to upgrade and is fine running an EOL Node.js, then they should be fine running an older elm-test as well. It’s not like the old version would stop working for them.

Reply:
What we should have done was to drop node 8 support when we released 0.19.1. We should definitely make sure that we drop support for enough node versions when we drop 0.19.2 so that we avoid this problem in the future.

> -   Upgrading Node.js is super easy these days anyway, IMO.

True

> -   And if someone doesn’t want to upgrade and is fine running an EOL Node.js, then they should be fine running an older elm-test as well. It’s not like the old version would stop working for them.

Also true.

Reply:
Copying in stuff from comments in another thread (so I can minimize the other one).

These are our choices:

1. Drop Node.js 8 support.
2. Fallback to this in Node.js 8: https://mothereff.in/regexpu#input=const+nonCapitalizedNamePart+%3D+%2F%5Cp%7BLl%7D%5B_%5Cd%5Cp%7BL%7D%5D*%2Fu%3B&dotAllFlag=1&unicodePropertyEscape=1
3. Use `[a-z][_\da-zA-Z]*` in Node.js 8.

Reply:
Could you make a separate PR to drop node 8 support? (Add node 15 to the matrix at the same time if you like)

Reply:
Sure!

Reply:
PR: https://github.com/rtfeldman/node-test-runner/pull/461